src/azul/indexer/index_service.py

```diff
@@ -346,7 +346,11 @@ def _read_contributions(self, tallies: CataloguedTallies) -> List[CataloguedCont
         hits = None
         if num_contributions <= page_size:
             log.info('Reading %i expected contribution(s) using search().', num_contributions)
-            response = es_client.search(index=index, body=query, size=page_size, doc_type=Document.type)
+            response = es_client.search(index=index,
+                                        body=query,
+                                        size=page_size,
+                                        doc_type=Document.type,
+                                        seq_no_primary_term=True)
```

Comment:
Still brittle coupling, Assumes that only contributions use internal versioning. If for some reason we switched aggregates to use internal versioning, too, things would break.

Reply:
https://github.com/DataBiosphere/azul/blob/3ee6f03ab1115275ec1dcd04af95cb20ff384148/src/azul/indexer/document.py#L591

Looking into this more.

Reply:
When aggregates are first created, the aggregate class is formed by the contribution, and the __init__ explicitly has the parameter `version=None` [passed to it.](https://github.com/DataBiosphere/azul/blob/0770e7c96e9a23b9e874745a2df28b29b64fe2a1/src/azul/indexer/index_service.py#L442)

The next time the old_aggregates are fetched, the relative documents for the tallies are obtained with `mget`
https://github.com/DataBiosphere/azul/blob/0770e7c96e9a23b9e874745a2df28b29b64fe2a1/src/azul/indexer/index_service.py#L295
`mget` returns multiple docs with responses that are similar to [get](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docs-get.html#get-source-filtering). The document objects have their applicable `seq_num` && `primary_term` within the response. This document is used as the parameter for the aggregate_cls.from_index() method, where it is extracted into the version tuple.